arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1188,14 +1190,18 @@ void RocksDBCollection::figuresSpecific(std::shared_ptr<arangodb::velocypack::Bu
 
 namespace {
 template<typename F>
-void reverseIdxOps(std::vector<std::shared_ptr<Index>> const& vector,
-                   std::vector<std::shared_ptr<Index>>::const_iterator& it,
+void reverseIdxOps(PhysicalCollection::IndexContainerType const& indexes,
+                   PhysicalCollection::IndexContainerType::const_iterator& it,
                    F&& op) {
-  while (it != vector.begin()) {
+  while (it != indexes.begin()) {
     it--;
     auto* rIdx = static_cast<RocksDBIndex*>(it->get());
-    if (rIdx->type() == Index::TRI_IDX_TYPE_IRESEARCH_LINK) {
-      std::forward<F>(op)(rIdx);
+    if (rIdx->needsReversal()) {
+      if (std::forward<F>(op)(rIdx).fail()) {
+        // best effort for reverse failed. Let`s trigger full rollback  
+        // or we will end up with inconsistent storage and indexes
+        THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL);
```

Comment:
we should never throw TRI_ERROR_INTERNAL without a corresponding message attached.
I suggest changing this to `THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "some descriptive message");`.